Extension refactored to adapt to GNOME 45

pratap-panabaka opened the pull request:
Hi,

I tried to adapt the extension from GNOME 44 code to GNOME 45.
I could not make the location in the Prefs window to choose from drop-down so used the spin button same as index selection.

![image](https://github.com/one-thing-gnome/one-thing/assets/40719899/9d8fc8b0-e4e6-4c5e-9b9d-2559cf61c194)

Thanks

Commit:
refactoring for gnome-shell v45

Commit:
update Prefs

jancborchardt reviewed (commented):
Looks nice, can’t test on GNOME 45 though. :) 

@PRATAP-KUMAR what do you think about using a settings cog icon on the right of the "Write One Thing …" input field for getting to settings instead of the dedicated line below? That would look a bit simpler, but I’m not sure if it is ok according to the GNOME HIG.

kra-mo commented:
"You can always access it in Extension Manager" should probably be a description, not part of the title. Extension Manager also isn't an official GNOME app, so it should say Extensions.

kra-mo commented:
@jancborchardt The HIG doesn't extend to the Shell, but I don't see why it would go against it. Having a simple icon for common actions like this is preferable to text, so I think it's fine in that regard.

Commit:
Code review comments incorporated

pratap-panabaka commented:
Hi, I tried to incorporate above discussion points.

![image](https://github.com/one-thing-gnome/one-thing/assets/40719899/0b1bf868-3ba6-4fe9-b363-45a1110158bb)

kra-mo committed:
Misc UI changes

kra-mo commented:
I've pushed some UI changes to the branch. This is what it looks like now:

![Screenshot from 2023-10-20 15-54-38](https://github.com/one-thing-gnome/one-thing/assets/93832451/388910cb-ea4e-47e0-aeeb-366ace17ef74)

![Screenshot from 2023-10-20 15-54-42](https://github.com/one-thing-gnome/one-thing/assets/93832451/4b812035-d1a7-49d7-8c49-fb299da20f87)

Otherwise LGTM

pratap-panabaka commented:
Looks Nice. If possible plz make changes to Location Prefs to Left, Center, Right. I am not able to achieve that.

Thanks

kra-mo committed:
Make location switcher grouped ToggleButtons

kra-mo commented:
@PRATAP-KUMAR Done.

![image](https://github.com/one-thing-gnome/one-thing/assets/93832451/9016b44e-e32f-4142-a22c-5c9fc63b1cdd)


Though I am not very proficient with JavaScript, so if you think you could clean the code up, go ahead.

kra-mo committed:
Update screenshots

Commit:
Add suport for emojis via Prefs

pratap-panabaka commented:
Hi,
I have updated Prefs to have option for Emojis via Adw.EntryRow.
Also added `Prefs` via Context menu along with copy, paste.

Any changes you feel, feel free to change and look to upload to gnome-extensions.

Thanks

![image](https://github.com/one-thing-gnome/one-thing/assets/40719899/09b5903d-cbf8-45ae-9b96-415be85bbd4a)

Commit:
update hint text for emojis

kra-mo committed:
Minor UX changes

kra-mo commented:
Having `<primary>,` open an emoji picker is standard across GNOME, so there really doesn't need to be text explaining it. I pushed a commit that removes it and updates the screenshot.

![Screenshot from 2023-10-21 13-07-50](https://github.com/one-thing-gnome/one-thing/assets/93832451/ea52b0bb-a62d-4c63-a97a-edb0ba274440)

pratap-panabaka commented:
That is nice. I hope the owner of extension can now try to submit it for review from gnome-extensions.

dantehemerson reviewed (commented):
@PRATAP-KUMAR @kra-mo Thanks for these changes!
I just leave a couple of comments:

dantehemerson reviewed (approved):
LGTM!

Commit:
git ignore gschemas.compiled and *.zip files remove _generated from metadata file

kra-mo commented:
@jancborchardt your verdict?

dantehemerson committed:
Untrack files in .gitignore

dantehemerson commented:
I sent the new version for review: https://extensions.gnome.org/review/47068

pratap-panabaka commented:
> I sent the new version for review: https://extensions.gnome.org/review/47068

Nice. Lets wait for the approval. Hope there wont be any comments from the reviewer. If anything please share.

pratap-panabaka commented:
Hi, I have renamed the branch master to main.

![image](https://github.com/one-thing-gnome/one-thing/assets/40719899/2d2df94a-04c7-4669-ac98-14995cb13b74)

```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

dantehemerson commented:
@PRATAP-KUMAR @kra-mo We received a comment in the review: https://extensions.gnome.org/review/47068

pratap-panabaka commented:
Ok, but the reviewer is saying about right panel, what if the user chooses to be at left or at the middle? do you have any idea @dantehemerson

pratap-panabaka commented:
@dantehemerson I have made the changes and pushed to main branch. Tested the extension. please have a look. Thanks

![image](https://github.com/one-thing-gnome/one-thing/assets/40719899/47a89ad2-dcb0-4778-bc33-1446cac8c812)

dantehemerson commented:
@PRATAP-KUMAR I don't have gnome 45, I cannot test. But looks good to me. I will send for review.

dantehemerson commented:
@PRATAP-KUMAR I just have this question: https://github.com/one-thing-gnome/one-thing/commit/c4f415f5c6ac03f89dd7ec2c51479674178f86c3#r131123642

pratap-panabaka commented:
@dantehemerson I have just tested with else statement and the extension is not working as expected. I understood what you mean. We may have to dig more if we want to go with else clause, I think `actorAddedSignal` is causing the issue. Shall we dig more on this or submit for review with the latest commit I made?
